CMakeLists.txt

```diff
@@ -9,14 +9,17 @@ if(APPLE)
   option(ZMQ_BUILD_FRAMEWORK "Build as OS X framework" ON)
 endif()
 
-if(WIN32)
-  option(WITH_TWEETNACL "Build with tweetnacl" OFF)
-else()
-  option(WITH_TWEETNACL "Build with tweetnacl" ON)
-endif()
+list(INSERT CMAKE_MODULE_PATH 0 "${CMAKE_SOURCE_DIR}")
+find_package(Sodium)
 
-if(WITH_TWEETNACL)
-  add_definitions(-DHAVE_TWEETNACL -DHAVE_LIBSODIUM)
+set(USE_TWEETNACL OFF)
+
+if(SODIUM_FOUND)
+  add_definitions(-DHAVE_LIBSODIUM)
+  include_directories(${SODIUM_INCLUDE_DIRS})
+else()
+  set(USE_TWEETNACL ON)
```

Comment:
So this breaks windows now when not using libsodium.

randombytes is an unresolved external, presumably tweetnacl has never worked on windows but now we are forcing its use.